secrets/store.go

```diff
@@ -149,5 +167,21 @@ func (s Store) GetCredentialSecret(path string) (CredentialSecret, error) {
 //
 // This function always returns nil error.
 func (s Store) GetVault() (Vault, error) {
+	if s.actual != nil {
+		return Vault{}, fmt.Errorf("GetVault cannot be used with Wrap'd Store")
+	}
 	return s.getSecrets().vault, nil
 }
+
+// A secretStore is an interface that is satisfied by both v0 and v2 secrets.
+type secretStore interface {
+	GetSimpleSecret(path string) (SimpleSecret, error)
+	GetVersionedSecret(path string) (VersionedSecret, error)
+	GetCredentialSecret(path string) (CredentialSecret, error)
+}
+
+// Wrap returns a Store that forwards all secret retrieval operations to the given store.
+// The returned Store cannot use middleware or provide its Vault information.
+func Wrap(store secretStore) *Store {
```

Comment:
this means we have to make a public api using an internal interface, just export `secretStore` interface?

I guess there's a possibility that we would add another common interface to both v0 and v2 and that will cause a breaking change to a public interface, but I think that possibility is pretty low at this point? if we really think that's possible then maybe just use `any` in the function signature and return `nil` if the `store` passed in does not satisfy the interface?